File: Projeto Didio Pizza/Didio Pizza Node/backend/src/service/estoqueService.js
import * as repository from '../repository/estoqueRepository.js';


export async function criar(item) {

    if (!item.nome || !item.quantidade || !item.unidade) {
        throw new Error('Todos os campos são obrigatórios.');
    }

    if (item.quantidade < 0) {
        throw new Error('Quantidade inválida.');
    }

    return await repository.inserir(item);
}


export async function listar() {
    return await repository.listar();
}


export async function atualizar(id, item) {

    if (!item.nome || !item.quantidade || !item.unidade) {
        throw new Error('Campos obrigatórios.');
    }

    let linhas = await repository.atualizar(id, item);

    if (linhas == 0) {
        throw new Error('Item não encontrado.');
    }
}


export async function remover(id) {

    let linhas = await repository.deletar(id);

    if (linhas == 0) {
        throw new Error('Item não encontrado.');
    }
}

File: Projeto Didio Pizza/Didio Pizza Springboot/backend/src/main/java/com/pizzaria/didiopizza/service/EstoqueService.java
package com.pizzaria.didiopizza.service;

import java.util.List;
import java.util.Map;

import org.springframework.stereotype.Service;

import com.pizzaria.didiopizza.repository.EstoqueRepository;

@Service
public class EstoqueService {
    private final EstoqueRepository repo;

    public EstoqueService(EstoqueRepository repo) {
        this.repo = repo;
    }

    private double num(Object v) {
        return v == null ? 0 : Double.parseDouble(String.valueOf(v));
    }

    public Integer criar(Map<String, Object> i) {
        if (i.get("nome") == null || i.get("quantidade") == null || i.get("unidade") == null)
            throw new RuntimeException("Todos os campos são obrigatórios.");
        if (num(i.get("quantidade")) < 0)
            throw new RuntimeException("Quantidade inválida.");
        return repo.inserir(i);
    }

    public List<Map<String, Object>> listar() {
        return repo.listar();
    }

    public void atualizar(Integer id, Map<String, Object> i) {
        if (i.get("nome") == null || i.get("quantidade") == null || i.get("unidade") == null)
            throw new RuntimeException("Campos obrigatórios.");
        if (repo.atualizar(id, i) == 0)
            throw new RuntimeException("Item não encontrado.");
    }

    public void remover(Integer id) {
        if (repo.deletar(id) == 0)
            throw new RuntimeException("Item não encontrado.");
    }
}
